src/charm.py

```diff
@@ -255,7 +277,16 @@ def ready_to_start(self) -> bool:
         return True
 
     def get_secret(self, scope: str, key: str) -> Optional[str]:
-        """Get TLS secret from the secret storage."""
+        """Get TLS secret from the secret storage.
+
+        Args:
+            scope: whether this secret is for a `unit` or `app`
+            key: the secret key name
+
+        Returns:
+            String of key value.
+            None if non-existent key
+        """
         if scope == "unit":
             return self.unit_peer_data.get(key, None)
         elif scope == "app":
```

Comment:
NIT: (personal preference, given you have `return` statements in the conditions)
```
if scope == "unit":
    return self.unit_peer_data.get(key)

if scope == "app":
    return self.app_peer_data.get(key)
    
raise RuntimeError("Unknown secret scope.")
```

Reply:
NIT: To remove the need of raising an exception here, it would make sense to pass an Enum value as the key instead of a string.
i.e: 
```
class Scope(Enum):
    APP = "app"
    UNIT = "unit"
    
    def __str__(self):
        return self.value
```

Reply:
I like the idea. I won't put it in this PR but will address in others for all 4 repos :)